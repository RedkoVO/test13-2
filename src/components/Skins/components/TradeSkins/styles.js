import CheckedIcon from '../../../../assets/images/svg/checked.svg'

export default () => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    flexWrap: 'wrap',
    width: '520px',
    padding: '25px',
    backgroundColor: 'rgba(0,0,0,0.08)',
    borderRadius: '15px',
    marginTop: '25px',
    boxSizing: 'border-box'
  },
  tradeSkinsTitle: {
    fontSize: '17px',
    textAlign: 'center',
    flexBasis: '100%',
    marginBottom: '25px'
  },
  tradeSkinsItemWrap: {
    width: '100%',
    height: '420px',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-around',
    flexWrap: 'wrap'
  },
  tradeSkinsItem: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.08)',
    borderRadius: '15px',
    width: '100px',
    height: '125px',
    padding: '5px',
    boxSizing: 'border-box',
    marginBottom: '10px'
  },
  skinSelected: {
    position: 'relative',
    width: '100px',
    height: '125px',
    borderRadius: '15px',

    '&:after': {
      position: 'absolute',
      content: `''`,
      width: '100%',
      height: '100%',
      background: `url(${CheckedIcon})`,
      backgroundRepeat: 'no-repeat',
      backgroundPosition: 'center 5%',
      backgroundSize: '50px',
      zIndex: 100
    }
  },
  skinsItemImg: {
    display: 'flex',
    alignItems: 'center',
    height: '50px',
    width: '85px',

    '& img': {
      maxWidth: '100%',
      maxHeight: '100%'
    }
  },
  skinsItemName: {
    fontSize: '10px',
    textAlign: 'center',
    marginTop: '7px'
  },
  skinsItemPrice: {
    fontSize: '14px',
    fontWeight: 500,
    textAlign: 'center',
    margin: '2px 0'
  },
  skinsItemSale: {
    fontSize: '10px',
    textAlign: 'center',
    fontWeight: 700,
    position: 'relative',

    '&:after': {
      position: 'absolute',
      content: `''`,
      width: '125%',
      height: '1px',
      backgroundColor: '#fff',
      left: '-12.5%',
      top: '50%'
    }
  },
  pageNavi: {
    minWidth: '100px',
    height: '30px',
    margin: '10px',
    marginBottom: '10px',
    padding: '0px',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    listStyleType: 'none',

    '& li': {
      margin: '0 1px',

      '&:last-child div': {
        marginRight: '0px'
      },

      '& div': {
        display: 'block',
        width: '100%',
        height: '100%',
        padding: '5px 7px',
        boxSizing: 'border-box',
        fontSize: '11px',
        fontWeight: 700,
        textDecoration: 'none',
        background: '#fff',
        transition: 'background 200ms linear',
        backgroundColor: 'rgba(0,0,0,0.18)',
        color: '#fff',

        '&.current': {
          background: 'rgba(255,255,255,0.18)',
          color: '#fff'
        },
        '&.disable': {
          background: 'rgba(255,255,255,0.18)',
          color: '#fff'
        },
        '&:hover': {
          background: 'rgba(255,255,255,0.18)',
          color: '#fff'
        }
      }
    }
  },
  previos: {
    borderTopLeftRadius: '10px',
    borderBottomLeftRadius: '10px'
  },
  next: {
    borderTopRightRadius: '10px',
    borderBottomRightRadius: '10px'
  }
})
