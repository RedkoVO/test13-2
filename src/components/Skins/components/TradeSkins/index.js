import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'
import cn from 'classnames'

import GunLogo from '../../../../assets/images/gun.png'

import styles from './styles'

const TradeSkins = ({ classes, skins }) => (
  <div className={classes.root}>
    <div className={classes.tradeSkinsTitle}>
      Select skins you want to sale:
    </div>
    <div className={cn(classes.tradeSkinsItemWrap, classes.current)}>
      {skins.map(item => (
        <div className={cn(classes.tradeSkinsItem, classes.skinSelected)} key={item.id}>
          <div className={classes.skinsItemImg}>
            <img src={GunLogo} alt="" />
          </div>
          <div className={classes.skinsItemName}>{item.title}</div>
          <div className={classes.skinsItemPrice}>{item.cost}</div>
          <div className={classes.skinsItemSale}>{item.discountsCost}</div>
        </div>
      ))}
    </div>

    <ul className={classes.pageNavi}>
      <li>
        <div className={cn(classes.previos, classes.disable)}>Prev</div>
      </li>
      <li>
        <div className={classes.current}>1</div>
      </li>
      <li>
        <div>2</div>
      </li>
      <li>
        <div className={classes.next}>Next</div>
      </li>
    </ul>
  </div>
)

TradeSkins.propTypes = {
  classes: PropTypes.object,
  skins: PropTypes.array
}

export default withStyles(styles)(TradeSkins)
