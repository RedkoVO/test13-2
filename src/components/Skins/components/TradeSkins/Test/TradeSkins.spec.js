import * as React from 'react'
import renderer from 'react-test-renderer'

import TradeSkins from '../'

const skinsArr = [
  {
    id: 1,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 2,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 3,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 4,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 5,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  }
]

it('renders correctly', () => {
  const tree = renderer.create(<TradeSkins skins={skinsArr} />).toJSON()
  expect(tree).toMatchSnapshot()
})
