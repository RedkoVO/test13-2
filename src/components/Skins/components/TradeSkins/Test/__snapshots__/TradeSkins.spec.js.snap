// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="TradeSkins-root-1"
>
  <div
    className="TradeSkins-tradeSkinsTitle-2"
  >
    Select skins you want to sale:
  </div>
  <div
    className="TradeSkins-tradeSkinsItemWrap-3"
  >
    <div
      className="TradeSkins-tradeSkinsItem-4 TradeSkins-skinSelected-5"
    >
      <div
        className="TradeSkins-skinsItemImg-6"
      >
        <img
          alt=""
          src="gun.png"
        />
      </div>
      <div
        className="TradeSkins-skinsItemName-7"
      >
        Gun: Good Weapon2 1.2
      </div>
      <div
        className="TradeSkins-skinsItemPrice-8"
      >
        $11.10 FN
      </div>
      <div
        className="TradeSkins-skinsItemSale-9"
      >
        $12.30
      </div>
    </div>
    <div
      className="TradeSkins-tradeSkinsItem-4 TradeSkins-skinSelected-5"
    >
      <div
        className="TradeSkins-skinsItemImg-6"
      >
        <img
          alt=""
          src="gun.png"
        />
      </div>
      <div
        className="TradeSkins-skinsItemName-7"
      >
        Gun: Good Weapon2 1.2
      </div>
      <div
        className="TradeSkins-skinsItemPrice-8"
      >
        $11.10 FN
      </div>
      <div
        className="TradeSkins-skinsItemSale-9"
      >
        $12.30
      </div>
    </div>
    <div
      className="TradeSkins-tradeSkinsItem-4 TradeSkins-skinSelected-5"
    >
      <div
        className="TradeSkins-skinsItemImg-6"
      >
        <img
          alt=""
          src="gun.png"
        />
      </div>
      <div
        className="TradeSkins-skinsItemName-7"
      >
        Gun: Good Weapon2 1.2
      </div>
      <div
        className="TradeSkins-skinsItemPrice-8"
      >
        $11.10 FN
      </div>
      <div
        className="TradeSkins-skinsItemSale-9"
      >
        $12.30
      </div>
    </div>
    <div
      className="TradeSkins-tradeSkinsItem-4 TradeSkins-skinSelected-5"
    >
      <div
        className="TradeSkins-skinsItemImg-6"
      >
        <img
          alt=""
          src="gun.png"
        />
      </div>
      <div
        className="TradeSkins-skinsItemName-7"
      >
        Gun: Good Weapon2 1.2
      </div>
      <div
        className="TradeSkins-skinsItemPrice-8"
      >
        $11.10 FN
      </div>
      <div
        className="TradeSkins-skinsItemSale-9"
      >
        $12.30
      </div>
    </div>
    <div
      className="TradeSkins-tradeSkinsItem-4 TradeSkins-skinSelected-5"
    >
      <div
        className="TradeSkins-skinsItemImg-6"
      >
        <img
          alt=""
          src="gun.png"
        />
      </div>
      <div
        className="TradeSkins-skinsItemName-7"
      >
        Gun: Good Weapon2 1.2
      </div>
      <div
        className="TradeSkins-skinsItemPrice-8"
      >
        $11.10 FN
      </div>
      <div
        className="TradeSkins-skinsItemSale-9"
      >
        $12.30
      </div>
    </div>
  </div>
  <ul
    className="TradeSkins-pageNavi-10"
  >
    <li>
      <div
        className="TradeSkins-previos-11"
      >
        Prev
      </div>
    </li>
    <li>
      <div>
        1
      </div>
    </li>
    <li>
      <div>
        2
      </div>
    </li>
    <li>
      <div
        className="TradeSkins-next-12"
      >
        Next
      </div>
    </li>
  </ul>
</div>
`;
